### PR TITLE
API Contract Verification page

### PR DESCRIPTION
**Pull request by hyd628**
### Description

Create a new page for API based contract verification tools

### Checklist

- [x] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects
- [x] If pages have been moved around, I have run the `move-pages.py` script to move the pages and update the image paths on the chinese repo
    - [x] After the script has been run, I have created an additional PR in `moonbeam-docs-cn`
- [x] If images have been added, I have run the `compress-images.py` script to compress the images.
- [x] If variables (in variables.yml) need to be updated (such as a name change), I have updated the `moonbeam-docs-cn` repo to use the new variables

### Corresponding PRs

Please link to any corresponding PRs here.

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [x] No additional PRs are required after the translations are done

#### Items to be Updated

Please list any of the items that will need to be added or deleted after the translations are done here.

### CODE REVIEW

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+
+## Using Moonscan API {: #using-moonscan-api } 
+
+[Moonscan](https://moonscan.io/){target=_blank} is an official fork of Etherscan that can be used to view and search on-chain data, and comes with a suite of developer tools and analytics to interact with data on Moonbeam networks. 
+
+Moonscan exposes a set of REST API endpoints similar to the [Etherscan API](https://docs.etherscan.io/){target=_blank} endpoints that can be used to verify smart contracts, retrieve verified contract ABI and source code, or otherwise interact with verified contracts on Moonbeam networks. 
+
+### Generating a Moonscan API Key {: #generating-a-moonscan-api-key }
+
+Before using the Moonscan API, you need to generate a Moonscan API key. Please follow the instructions in [the key generation section](builders/build/eth-api/verify-contracts/etherscan-plugins/#generating-a-moonscan-api-key){target=_blank} of the Etherscan plug-in verification page, as the API keys generated are used for both.
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:24]:
```suggestion
Before using the Moonscan API, you need to generate a Moonscan API key. Please follow the instructions in [the key generation section](/builders/build/eth-api/verify-contracts/etherscan-plugins/#generating-a-moonscan-api-key){target=_blank} of the Etherscan plug-in verification page, as the API keys generated are used for both.
```

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+    ```
+
+=== "Moonbase Alpha"
+    ```
+    https://api-moonbase.moonscan.io/api
+    ```
+
+### Verify Source Code {: #verify-source-code }
+
+To verify a deployed contract's source code using the Moonscan API, you must form a POST request containing all the relevant contract creation information, and send the request to Moonscan's REST API. The following is sample code using Javascript:
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:47]:
```suggestion
To verify a deployed contract's source code using the Moonscan API, you must form a POST request containing all the relevant contract creation information, and send the request to Moonscan's REST API. The following is sample code using JavaScript:
```

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+
+=== "Moonbeam"
+    ```javascript
+    //Submit Source Code for Verification
+    $.ajax({
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:52]: is this jquery?

Comment (eshaben) [reply builders/build/eth-api/verify-contracts/api-verification.md:52]: there should be a space after all of these `//` (for example: `// Only POST supported` and all comments should start with capitalized letter

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+
+Upon successful submission, a GUID will be returned as a part of the result. This GUID can be used to check for the submission status. 
+
+=== "Moonbeam"
+    ```bash
+    curl https://api-moonbeam.moonscan.io/api
+            ?module=contract
+            &action=checkverifystatus
+            &guid=GUIDFromResponse
+            &apikey=YourApiKeyToken
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:198]: to be consistent across the docs site, i think this should be changed to GUID-FROM-RESPONSE-HERE and YOUR-API-KEY-TOKEN-HERE

Comment (eshaben) [reply builders/build/eth-api/verify-contracts/api-verification.md:198]: same with following examples

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+    ```bash
+    https://sourcify.dev/server
+    ```
+
+=== "Staging"
+    ```bash
+    https://staging.sourcify.dev/server
+    ```
+
+### Moonbeam Network Chain ID's
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:299]:
```suggestion
### Moonbeam Network Chain ID's {: #moonbeam-network-chain-ids }
```

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+
+Sourcify uses Chain ID's to identify the target network(s) for the request. The chain ID's of Moonbeam networks are as follows:
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:301]:
```suggestion
Sourcify uses chain ID's to identify the target network(s) for the request. The chain ID's of Moonbeam networks are as follows:
```

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+
+Sourcify supports two types of verification match results. 
+
+Full matches (sometimes referred as perfect matches) refer to the cases when the bytecode of the deployed contract is byte-by-byte the same as compilation output of the given source code files under the compilation settings defined in the metadata file.
+
+Partial matches refer to cases when the deployed bytecode of the onchain contract match the bytecode resulting from the recompilation with the metadata and the source files except the metadata hash. For partial matches, the deployed contract and the given source code and metadata are functionally the same, but there are differences in source code comments, variable names, or other metadata fields such as source paths.
+
+### Verify Contract {: #verify-contract }
+
+A POST request is used to verify a contract on Sourcify. The following is sample code using Javascript.
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:328]:
```suggestion
A POST request is used to verify a contract on Sourcify. The following is sample code using JavaScript.
```

File: builders/build/eth-api/verify-contracts/api-verification.md [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+                "file1-name.sol": $('#file_1').val(), // Contract source file 1
+                "file2-name.sol": $('#file_2').val() // Contract source file 2
+                //...
+            },
+            "chosenContract": 1 // Optional. Index of the contract, if the provided files contain multiple metadata files  
+        },
+        success: function (result) {
+            console.log(result);
+            if (result.status == "perfect") {
+                // Perfect Match
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/api-verification.md:351]: the second word here doesn't need to be capitalized... same applies throughout

File: builders/build/eth-api/verify-contracts/block-explorers.md [hunk 9]
```diff
@@ -9,9 +9,9 @@ description: Learn how to verify smart contracts on Moonbeam-based networks usin
 
 ## Introduction {: #introduction } 
 
-Verifying smart contracts on a block explorer is great way of improving the transparency and security of deployed smart contracts on Moonbeam. Users can directly view the source code for verified smart contracts, and for some block explorers, can also directly interact with the contract's public methods through the block explorer's interface. 
+Verifying smart contracts on a block explorer is a great way of improving the transparency and security of deployed smart contracts on Moonbeam. Users can directly view the source code for verified smart contracts, and for some block explorers, can also directly interact with the contract's public methods through the block explorer's interface. 
 
-This page will outline the steps for verifying smart contracts on Moonbeam networks.
+This page will outline the steps for verifying smart contracts on Moonbeam networks through block explorers.
```
Comment (eshaben) [inline builders/build/eth-api/verify-contracts/block-explorers.md:14]: thanks for cleaning this page up!

**Review by eshaben — CHANGES_REQUESTED** [review]
Looks goooooood! 🙂  I see this is in draft but I went through and did an initial review for now

File: builders/build/eth-api/verify-contracts/api-verification.md (hunk 3 above)
Comment (hyd628) [reply builders/build/eth-api/verify-contracts/api-verification.md:52]: > is this jquery?

Maybe, I'm not too clear on the difference between jquery javascript and AJAX lol. The javascript linter(?) seems to pick up the syntax.

**Comment by hyd628** [comment]
@eshaben thanks for the feedback, Erin. I've incorporated the changes.

Also added the icon image and banner, so will take this out of draft.

**Review by eshaben — COMMENTED** [review]
LGTM ✨ 

I made some changes to the code snippets to use axios instead of jquery b/c we really only need an http client for this. removed the aspects where the code snippet was getting values from a UI form to make it more usable for a wider audience and i felt like there was some context missing when the code snippet is grabbing from a UI but we don't have an example UI

**Pull request merged**